identity/app/actions/AuthenticatedActions.scala

```diff
@@ -68,105 +64,117 @@ class AuthenticatedActions(
+
+          case (true, false) =>
+            Future.successful(Left(sendUserToAllConsentsJourney(request)))
+
+          case (true, true) =>
+            newsletterService.subscriptions(request.user.getId, idRequestParser(request).trackingData).map {
+              emailFilledForm =>
+                if (newsletterService.getV1EmailSubscriptions(emailFilledForm).isEmpty)
+                  Right(request)
+                else
+                  Left(sendUserToNewslettersConsentsJourney(request))
+              }
+        }
 
-  def recentlyAuthenticatedRefiner: ActionRefiner[AuthRequest, AuthRequest] = new ActionRefiner[AuthRequest, AuthRequest] {
-    override val executionContext = ec
+      private def userHasRepermissioned(request: AuthRequest[_]): Boolean =
+        request.user.statusFields.hasRepermissioned.contains(true)
+
+      private def userEmailValidated(request: AuthRequest[_]): Boolean =
+        request.user.statusFields.isUserEmailValidated
+    }
+
+  private def recentlyAuthenticatedRefiner: ActionRefiner[AuthRequest, AuthRequest] =
+    new ActionRefiner[AuthRequest, AuthRequest] {
+      override val executionContext = ec
+
+      def refine[A](request: AuthRequest[A]) = checkRecentAuthenticationAndRedirect(request)
+    }
 
-    def refine[A](request: AuthRequest[A]) = checkRecentAuthenticationAndRedirect(request)
-  }
   // Play will not let you set up an ActionBuilder with a Refiner hence this empty actionBuilder to set up Auth
-  def noOpActionBuilder: DefaultActionBuilder = DefaultActionBuilder(anyContentParser)
+  private def noOpActionBuilder: DefaultActionBuilder = DefaultActionBuilder(anyContentParser)
 
-  def authAction: ActionBuilder[AuthRequest, AnyContent] =
-    noOpActionBuilder andThen authRefiner
+  /** SC_GU_U cookie present */
+  def fullAuthAction: ActionBuilder[AuthRequest, AnyContent] =
+    noOpActionBuilder andThen fullAuthRefiner
 
-  def authActionWithUser: ActionBuilder[AuthRequest, AnyContent] =
-    authAction andThen apiVerifiedUserRefiner
+  /** SC_GU_U cookie present and user retrieved from IDAPI */
+  def fullAuthWithIdapiUserAction: ActionBuilder[AuthRequest, AnyContent] =
+    fullAuthAction andThen retrieveUserFromIdapiRefiner
 
-  def recentlyAuthenticated: ActionBuilder[AuthRequest, AnyContent] =
-    authAction andThen recentlyAuthenticatedRefiner andThen apiVerifiedUserRefiner
+  /** Recently obtained SC_GU_U cookie and user retrieved from IDAPI */
+  def recentFullAuthWithIdapiUserAction: ActionBuilder[AuthRequest, AnyContent] =
+    fullAuthAction andThen recentlyAuthenticatedRefiner andThen retrieveUserFromIdapiRefiner
 
-  def authWithConsentRedirectAction: ActionBuilder[AuthRequest, AnyContent] =
-    authWithRPCookie andThen apiUserShouldRepermissionRefiner
+  /** Auth with at least SC_GU_RP, that is, auth with SC_GU_U or else SC_GU_RP, and user retrieved from IDAPI */
+  def consentAuthWithIdapiUserAction: ActionBuilder[AuthRequest, AnyContent] =
```

Comment:
Not sold on this name, maybe a more generic `magicLinkAuthWithIdapiUserAction` would be better? thoughts?